### PR TITLE
 Missing escape character on regex in YAML

### PR DESCRIPTION
**Pull request by fhalim**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by slim-bean** [comment]
Adding this escape would break the regex, see [this example](https://regex101.com/r/D42vhB/1)

**Comment by slim-bean** [comment]
I know it's annoying the way the yaml parses it, but I think it better for the regex to be correct?

**Comment by fhalim** [comment]
I felt that it'd be nice for to be able to copy/paste the YAML for beginners. As it stands, using this YAML verbatim with `promtail` shows a semi-cryptic message about an invalid escape character.

File: docs/logentry/processing-log-lines.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ scrape_configs:
       selector: '{name="promtail"}'
       stages:
       - regex:
-          expression: ".*level=(?P<level>[a-zA-Z]+).*ts=(?P<timestamp>[T\d-:.Z]*).*component=(?P<component>[a-zA-Z]+)"
+          expression: ".*level=(?P<level>[a-zA-Z]+).*ts=(?P<timestamp>[T\\d-:.Z]*).*component=(?P<component>[a-zA-Z]+)"
```
Comment (slim-bean) [inline docs/logentry/processing-log-lines.md]:
```suggestion
          expression: '.*level=(?P<level>[a-zA-Z]+).*ts=(?P<timestamp>[T\d-:.Z]*).*component=(?P<component>[a-zA-Z]+)'
```

**Comment by slim-bean** [comment]
ah you're right, and yes I completely agree that we want to be able to copy and paste the config.

Though rather than adding an escape character which I think will break that regex, I think we should instead replace the double quotes with single quotes, please see my inline suggestion.

File: docs/logentry/processing-log-lines.md (hunk 1 above)
Comment (fhalim) [reply docs/logentry/processing-log-lines.md]: That's perfect!

**Comment by slim-bean** [comment]
Can you update your PR with the suggested change to escape with single quotes and remove the backslash?  (I don't have a way to apply the suggested change and you should get credit for your work :) )

**Comment by fhalim** [comment]
> Can you update your PR with the suggested change to escape with single quotes and remove the backslash? (I don't have a way to apply the suggested change and you should get credit for your work :) )

Thanks! Done.

**Comment by slim-bean** [comment]
Looks great, Thanks!

**Pull request merged**